packages/TorneloScoresheet/src/pages/GraphicalRecording/GraphicalRecording.tsx

```diff
@@ -171,18 +173,32 @@ const GraphicalRecording: React.FC = () => {
     : [];
 
   // Button Functions
-  const handleConfirmWinner = (signature: string) => {
+
+  const handleConfirmWinner = (signatureInput: string) => {
+    const signatureArray = [...playerSignatures, signatureInput];
     if (!graphicalRecordingMode || !goToEndGame) {
       return;
     } else {
-      goToEndGame({
-        winner: selectedWinner?.color ?? null,
-        signature: signature,
-        gamePgn: pgn,
-      });
+      if (signatureArray.length == 2) {
+        setPlayerSignatures(() => []);
+        goToEndGame({
+          winner: selectedWinner?.color ?? null,
+          signature: signatureArray,
+          gamePgn: pgn,
+        });
+      }
     }
   };
 
+  const handleConfirmFirstSignature = (signatureInput: string) => {
+    setPlayerSignatures(playerSignatures => [
+      ...playerSignatures,
+      signatureInput,
+    ]);
+    setShowFirstSignature(false);
+    setShowSecondSignature(true);
+  };
+
```

Comment:
This works but I think we can come up with something slightly more elegant, if we have a record<PlayerColor, string|undefined> for the signatures, I think we can avoid having 2 separate handle confirm functions and 2 separate Signature components